### PR TITLE
Added more tests to the PGPNP signer

### PR DESCRIPTION
**Pull request by carterqw2**
### Description

More tests covering faulty paths with invalid inputs.

### CODE REVIEW

File: packages/phone-number-privacy/signer/test/key-management/google-key-provider.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    expect.assertions(1)
+    await expect(provider.fetchPrivateKeyFromStore()).rejects.toThrow()
+  })
+
+  it('unitialized provider throws', async () => {
+    accessSecretVersion.mockResolvedValue(mockResponse)
+
+    const provider = new GoogleKeyProvider()
+    expect.assertions(1)
+    await expect(() => provider.getPrivateKey()).toThrow()
```
Comment (jmrossy) [inline packages/phone-number-privacy/signer/test/key-management/google-key-provider.test.ts]: Does this need to be async?

**Review by jmrossy — APPROVED** [review]
Excellent! Just one minor thing

File: packages/phone-number-privacy/signer/test/key-management/google-key-provider.test.ts (hunk 1 above)
Comment (carterqw2) [reply packages/phone-number-privacy/signer/test/key-management/google-key-provider.test.ts]: No, it doesn't, good catch.

File: packages/phone-number-privacy/signer/test/key-management/google-key-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  it('handles errors correctly', async () => {
+    accessSecretVersion.mockResolvedValue(emptyMockResponse)
+
+    const provider = new GoogleKeyProvider()
+    expect.assertions(1)
+    await expect(provider.fetchPrivateKeyFromStore()).rejects.toThrow()
+  })
+
+  it('unitialized provider throws', () => {
```
Comment (codyborn) [inline packages/phone-number-privacy/signer/test/key-management/google-key-provider.test.ts:41]:
```suggestion
  it('uninitialized provider throws', () => {
```

**Review by codyborn — APPROVED** [review]

**Pull request merged**